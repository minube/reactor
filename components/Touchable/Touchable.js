import React from 'react';
import { Platform, TouchableNativeFeedback, TouchableOpacity } from 'react-native';

export default ({ ...inherit }) => (
  Platform.OS !== 'android'
    ? <TouchableOpacity {...inherit} />
    : <TouchableNativeFeedback useForeground {...inherit} />
);
