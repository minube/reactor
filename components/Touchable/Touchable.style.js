import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { BORDER_RADIUS, COLOR } = THEME;

export default StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: 'transparent',
    borderRadius: BORDER_RADIUS,
    overflow: 'hidden',
  },

  mask: {
    backgroundColor: COLOR.WHITE,
    zIndex: 0,
  },
});
