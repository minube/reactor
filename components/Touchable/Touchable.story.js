import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import {
  color, boolean, object, number,
} from '@storybook/addon-knobs/react';

import Text from '../Text';
import Touchable from './Touchable';

const LIPSUM = "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";
const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10 };

const Child = () => (
  <Text
    pointerEvents="none"
    style={{ backgroundColor: 'rgba(0,0,0,0.25)', padding: 10 }}
  >
    Hello World
  </Text>
);

storiesOf('✅ Touchable', module)
  .add('default', () => (
    <Touchable />
  ))
  .add('children', () => (
    <Touchable>
      <Child />
    </Touchable>
  ))
  .add('⚡ onPress', () => (
    <Touchable onPress={action('Touchable.onPress()')}>
      <Child />
    </Touchable>
  ))
  .add('rippleColor', () => (
    <Touchable onPress={() => {}} rippleColor="red">
      <Child />
    </Touchable>
  ))
  .add('containerBorderRadius', () => (
    <Touchable onPress={() => {}} containerBorderRadius={48}>
      <Child />
    </Touchable>
  ))
  .add('style', () => (
    <Touchable onPress={() => {}} style={STYLE}>
      <Child />
    </Touchable>
  ))
  .add('🏀 Playground', () => (
    <Touchable
      onPress={action('Touchable.onPress()')}
      rippleColor={color('rippleColor', undefined)}
      style={object('style', STYLE)}
    >
      <Child />
    </Touchable>
  ));
