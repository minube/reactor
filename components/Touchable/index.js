import { createElement } from 'react';
import { ENV } from '../../common';
import Touchable from './Touchable';

export default props => ENV.IS_SERVER ? props.children : createElement(Touchable, props); // eslint-disable-line
