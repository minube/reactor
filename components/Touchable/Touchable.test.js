import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Touchable from './Touchable';

describe('<Touchable>', () => {
  it('renders', () => {
    const tree = renderer.create(<Touchable><View /></Touchable>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Touchable><View /></Touchable>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

