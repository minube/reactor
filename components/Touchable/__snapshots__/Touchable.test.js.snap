// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Touchable> renders 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Touchable> when {children} 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello World
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Touchable> when {containerBorderRadius} 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 48,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Touchable> when {rippleColor} 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,255,0,0.25)",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Touchable> when {rippleDelay} 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<Touchable> when {style} 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="none"
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(0,255,0,0.25)",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;
